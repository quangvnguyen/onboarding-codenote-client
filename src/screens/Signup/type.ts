import { ISignUpResult } from 'amazon-cognito-identity-js';
import * as H from 'history';

export interface ISignupStates {
  password: string;
  email: string;
  confirmPassword: string;
  confirmationCode: string;
  isLoading: boolean;
  newUser: ISignUpResult;
}

export interface ISignupProps {
  userHasAuthenticated: (isAuthenticated: boolean) => void;
  history: H.History;
}

export interface IUser {
  username: string;
  password: string;
}