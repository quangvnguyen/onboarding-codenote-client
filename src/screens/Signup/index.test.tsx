import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import Signup from './index';

const mockStore = configureMockStore();
const store = mockStore({
  authenticate: {}
});

describe('Signup Component', () => {
  it('renders without crashing', () => {
    const wrapper = mount(
      <Signup store={store} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders confirm form', () => {
    const wrapper = mount(
      <Signup store={store} />
    );
    wrapper.setState({
      newUser: {},
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders when form validated', () => {
    const wrapper = mount(
      <Signup store={store} />
    );
    wrapper.setState({
      email: 'email@example.com',
      password: 'password',
      confirmPassword: 'password',
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders when handleSubmit function called', () => {
    const wrapper = mount(
      <Signup store={store} />
    );

    wrapper.find('form').simulate('submit');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders when handleSubmit function in confirm form called', () => {
    const wrapper = mount(
      <Signup store={store} />
    );
    wrapper.setState({
      newUser: {},
    });

    wrapper.find('form').simulate('submit');
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should handleChange function in email fields called', () => {
    const event = {target: {id: 'email', value: 'email'}};
    const wrapper = mount(
      <Signup store={store} />
    );

    wrapper.find('#email').simulate('change', event);
    expect(wrapper.state().email).toEqual('email');
  });

  it('should handleChange function in password fields called', () => {
    const event = {target: {id: 'password', value: 'password'}};

    const wrapper = mount(
      <Signup store={store} />
    );
    wrapper.find('#password').simulate('change', event);
    expect(wrapper.state().password).toEqual('password');
  });

  it('should handleChange function in confirmPassword fields called', () => {
    const event = {target: {id: 'confirmPassword', value: 'confirmPassword'}};

    const wrapper = mount(
      <Signup store={store} />
    );
    wrapper.find('#confirmPassword').simulate('change', event);
    expect(wrapper.state().confirmPassword).toEqual('confirmPassword');
  });
})