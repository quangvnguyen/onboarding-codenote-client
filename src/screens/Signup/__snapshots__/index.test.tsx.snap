// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component renders confirm form 1`] = `
<Signup
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="Signup"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="confirmationCode"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="confirmationCode"
            >
              Confirmation Code
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="tel"
            value=""
          >
            <input
              autoFocus={true}
              className="form-control"
              id="confirmationCode"
              onChange={[Function]}
              type="tel"
              value=""
            />
          </FormControl>
          <HelpBlock
            bsClass="help-block"
          >
            <span
              className="help-block"
            >
              Please check your email for the code.
            </span>
          </HelpBlock>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={true}
        isLoading={false}
        loadingText="Verifying…"
        text="Verify"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={true}
            type="submit"
          >
            Verify
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Signup>
`;

exports[`Signup Component renders when form validated 1`] = `
<Signup
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="Signup"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="email"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="email"
            >
              Email
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="email"
            value="email@example.com"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="email"
              onChange={[Function]}
              type="email"
              value="email@example.com"
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="password"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="password"
            >
              Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value="password"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              type="password"
              value="password"
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="confirmPassword"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="confirmPassword"
            >
              Confirm Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value="password"
          >
            <input
              className="form-control"
              id="confirmPassword"
              onChange={[Function]}
              type="password"
              value="password"
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={false}
        isLoading={false}
        loadingText="Signing up…"
        text="Signup"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={false}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={false}
            type="submit"
          >
            Signup
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Signup>
`;

exports[`Signup Component renders when handleSubmit function called 1`] = `
<Signup
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="Signup"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="email"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="email"
            >
              Email
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="email"
            value=""
          >
            <input
              autoFocus={true}
              className="form-control"
              id="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="password"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="password"
            >
              Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value=""
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="confirmPassword"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="confirmPassword"
            >
              Confirm Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value=""
          >
            <input
              className="form-control"
              id="confirmPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={true}
        isLoading={true}
        loadingText="Signing up…"
        text="Signup"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={true}
            type="submit"
          >
            <Glyphicon
              bsClass="glyphicon"
              className="spinning"
              glyph="refresh"
            >
              <span
                className="spinning glyphicon glyphicon-refresh"
              />
            </Glyphicon>
            Signing up…
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Signup>
`;

exports[`Signup Component renders when handleSubmit function in confirm form called 1`] = `
<Signup
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="Signup"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="confirmationCode"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="confirmationCode"
            >
              Confirmation Code
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="tel"
            value=""
          >
            <input
              autoFocus={true}
              className="form-control"
              id="confirmationCode"
              onChange={[Function]}
              type="tel"
              value=""
            />
          </FormControl>
          <HelpBlock
            bsClass="help-block"
          >
            <span
              className="help-block"
            >
              Please check your email for the code.
            </span>
          </HelpBlock>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={true}
        isLoading={true}
        loadingText="Verifying…"
        text="Verify"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={true}
            type="submit"
          >
            <Glyphicon
              bsClass="glyphicon"
              className="spinning"
              glyph="refresh"
            >
              <span
                className="spinning glyphicon glyphicon-refresh"
              />
            </Glyphicon>
            Verifying…
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Signup>
`;

exports[`Signup Component renders without crashing 1`] = `
<Signup
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="Signup"
  >
    <form
      onSubmit={[Function]}
    >
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="email"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="email"
            >
              Email
            </label>
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="email"
            value=""
          >
            <input
              autoFocus={true}
              className="form-control"
              id="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="password"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="password"
            >
              Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value=""
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <FormGroup
        bsClass="form-group"
        bsSize="large"
        controlId="confirmPassword"
      >
        <div
          className="form-group form-group-lg"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
              htmlFor="confirmPassword"
            >
              Confirm Password
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value=""
          >
            <input
              className="form-control"
              id="confirmPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormControl>
        </div>
      </FormGroup>
      <LoaderButton
        block={true}
        bsSize="large"
        disabled={true}
        isLoading={false}
        loadingText="Signing up…"
        text="Signup"
        type="submit"
      >
        <Button
          active={false}
          block={true}
          bsClass="btn"
          bsSize="large"
          bsStyle="default"
          className="LoaderButton "
          disabled={true}
          type="submit"
        >
          <button
            className="LoaderButton  btn btn-lg btn-default btn-block"
            disabled={true}
            type="submit"
          >
            Signup
          </button>
        </Button>
      </LoaderButton>
    </form>
  </div>
</Signup>
`;
